langchain/src/cache/momento.ts

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable no-instanceof/no-instanceof */
+import {
+  ICacheClient,
+  CacheGet,
+  CacheSet,
+  CreateCache,
+  InvalidArgumentError,
+} from "@gomomento/sdk";
+
+import { BaseCache, Generation } from "../schema/index.js";
+import { getCacheKey } from "./base.js";
+
+/**
+ * The settings to instantiate the Momento standard cache.
+ */
+export interface MomentoCacheProps {
+  /**
+   * The Momento cache client.
+   */
+  client: ICacheClient;
+  /**
+   * The name of the cache to use to store the data.
+   */
+  cacheName: string;
+  /**
+   * The time to live for the cache items. If not specified,
+   * the cache client default is used.
+   */
+  ttlSeconds?: number;
+  /**
+   * If true, ensure that the cache exists before returning.
+   * If false, the cache is not checked for existence.
+   * Defaults to true.
+   */
+  ensureCacheExists?: true;
+}
+
+/**
+ * A cache that uses Momento as the backing store.
+ * See https://gomomento.com.
+ */
+export class MomentoCache extends BaseCache {
+  private client: ICacheClient;
+
+  private readonly cacheName: string;
+
+  private readonly ttlSeconds?: number;
+
+  private constructor(props: MomentoCacheProps) {
+    super();
+    this.client = props.client;
+    this.cacheName = props.cacheName;
+
+    this.validateTtlSeconds(props.ttlSeconds);
+    this.ttlSeconds = props.ttlSeconds;
+  }
+
+  /**
+   * Create a new standard cache backed by Momento.
+   *
+   * @param {MomentoCacheProps} props The settings to instantiate the cache.
+   * @param {ICacheClient} props.client The Momento cache client.
+   * @param {string} props.cacheName The name of the cache to use to store the data.
+   * @param {number} props.ttlSeconds The time to live for the cache items. If not specified,
+   * the cache client default is used.
+   * @param {boolean} props.ensureCacheExists If true, ensure that the cache exists before returning.
+   * If false, the cache is not checked for existence. Defaults to true.
+   * @throws {@link InvalidArgumentError} if {@link props.ttlSeconds} is not strictly positive.
+   * @returns The Momento-backed cache.
+   */
+  public static async CreateAsync(
```

Comment:
We've standardized `fromX` for factory methods rather than capitals (e.g. `fromProps()` or `fromOptions()` - would you mind renaming?

Reply:
Refactored in https://github.com/hwchase17/langchainjs/pull/1416/commits/de103e65f50d8d4a563feba4564d56c90942bf68